cxx-qt-gen/src/parser/qobject.rs

```diff
@@ -0,0 +1,479 @@
+/// This has initial splitting of [syn::Item]'s into relevant blocks, other phases will
+/// then mutate these [syn::Item]'s for generation purposes.
+#[derive(Default)]
+pub struct ParsedQObject {
+    /// This is the Data struct that currently stores the properties for the QObject
+    ///
+    /// In the future this will be removed
+    pub data_struct: Option<ItemStruct>,
+    /// This is the RustObj struct that stores the invokables for the QObject
+    pub rust_struct: Option<ItemStruct>,
+    /// This is the Signals enum that defines the Q_SIGNALS for the QObject
+    pub signal_enum: Option<ItemEnum>,
+    /// This is the list of methods that need to be implemented on the C++ object in Rust
+    ///
+    /// TODO: in the future we will have a ParsedMethod this will have a field to say whether the
+    /// method is invokable or not.
+    /// Ones marked as invokable will need to be exposed to C++ as Q_INVOKABLE
+    pub methods: Vec<ImplItem>,
+    /// This is update request handler for the QObject
+    ///
+    /// In the future this may be removed
+    pub update_requester_handler: Option<ItemImpl>,
+    /// Items that we don't need to generate anything for CXX or C++
+    /// eg impls on the Rust object or Default implementations
+    pub others: Vec<Item>,
+}
+
+#[derive(Default)]
+pub struct ParsedCxxQtData {
+    /// Map of the QObject's defined in the module that need generation
+    pub qobjects: HashMap<Ident, ParsedQObject>,
+    /// Any `use` statements end up in the CXX-Qt generated module
+    pub uses: Vec<Item>,
+}
+
+impl ParsedCxxQtData {
+    /// Find the QObject's within the module and adds the keys into the QObject's HashMap
+    pub fn find_qobject_keys(&mut self, items: &[Item]) -> Result<()> {
+        for item in items {
+            match item {
+                Item::Struct(s) => {
+                    // TODO: instead find the cxx_qt::qobject macro
+                    // and support multiple QObjects in one block
+                    if s.ident == "RustObj" {
+                        // TODO: for now we only support one qobject per block
+                        if !self.qobjects.is_empty() {
+                            return Err(Error::new(
+                                s.span(),
+                                "Only on RustObj struct is supported per mod",
+                            ));
+                        }
+
+                        // Note that we assume a compiler error will occur later
+                        // if you had two structs with the same name
+                        self.qobjects
+                            .insert(s.ident.clone(), ParsedQObject::default());
+                    }
+                }
+                _others => {}
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Determine if the given [syn::Item] is a CXX-Qt related item
+    /// If it is then add the [syn::Item] into QObject's list
+    /// Otherwise return the [syn::Item] if it's not relevant
```

Comment:
```suggestion
    /// Determine if the given [syn::Item] is a CXX-Qt related item
    /// If it is then add the [syn::Item] into the qobjects HashMap
    /// Otherwise return the [syn::Item] to pass through to cxx
```

Reply:
done